src/timezones.c

```diff
@@ -0,0 +1,51 @@
+/*
+ * This file and its contents are licensed under the Apache License 2.0.
+ * Please see the included NOTICE for copyright information and
+ * LICENSE-APACHE for a copy of the license.
+ */
+
+#include <postgres.h>
+#include <port.h>
+#include <pgtime.h>
+#include <datatype/timestamp.h>
+#include <utils/timestamp.h>
+#include <access/xact.h>
+#include "timezones.h"
+
+/* Checks if the given TZ name is valid. */
+bool
+ts_is_valid_timezone_name(const char *tz_name)
+{
+	pg_tz *tz;
+	int tzoff;
+	struct pg_tm tm;
+	fsec_t fsec;
+	const char *abbrev;
+	bool found = false;
+	TimestampTz now = GetCurrentTransactionStartTimestamp();
+	pg_tzenum *tzenum = pg_tzenumerate_start();
+
+	for (;;)
+	{
+		tz = pg_tzenumerate_next(tzenum);
+		if (!tz)
+			break;
+
+		/* Convert now() to time in this TZ and skip if conversion fails. */
+		if (timestamp2tm(now, &tzoff, &tm, &fsec, &abbrev, tz) != 0)
+			continue;
+
+		/* Skip long abbreviations the same way pg_timezone_names() does. */
+		if (abbrev && strlen(abbrev) > 31)
+			continue;
+
```

Comment:
I think we can remove those lines completely since we already do `strcmp(tz_name, abbrev)`

Reply:
I don't think we can do it. This will make the behavior different from the PostgreSQL one. If there is a long abbreviation in the database (which can be the internal PostgreSQL one, or the system TZif file, depending on the configure flags) we will allow specifying it, while PostgreSQL will reject it. The rest of the TZ support in the CAGGs will break.

Reply:
On second thought the previous line checks that the TZ will not be rejected, so I think you are right.